parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -485,4 +486,80 @@ process_benchmarks() {
   set -x
 }
 
+#return proper release metadata for the package based on the os and osVersion
+#e.g., For os UBUNTU 18.04, if there is a release "r1804" defined in components.json, then return "r1804"
+#Otherwise return "current"
+returnRelease() {
+    local package="$1"
+    local os="$2"
+    local osVersion="$3"
+    local release="current"
+    local osVersionWithoutDot=$(echo "${osVersion}" | sed 's/\.//g')
+    #For UBUNTU, if $osVersion is 18.04 and "r1804" is also defined in components.json, then $release is set to "r1804"
+    #Similarly for 20.04 and 22.04. Otherwise $release is set to .current.
+    #For MARINER, the release is always set to "current" now.
+    if [[ "${os}" != "${UBUNTU_OS_NAME}" ]]; then
+        echo "${release}"
+        return
+    fi
+    if [[ $(echo "${package}" | jq ".downloadURIs.ubuntu.\"r${osVersionWithoutDot}\"") != "null" ]]; then
+        release="\"r${osVersionWithoutDot}\""
```

Comment:
are we sure this works properly? seems like it would be assigning to the locally-scoped `release` var

Reply:
This function will return a string by `echo "${release}"` and the caller will capture it. I have a unit test to cover this so it's working as expected. I run also run a complete abe2e to ensure.

Reply:
so one thing we can do here is assign the output to a non-local variable within the function, then the caller will actually have access to it, that's actually how we do it in our release scripts

I'd personally prefer that over the caller having to capture stdout

Reply:
Updated.
Though I personally like to echo it so that we don't need so many global variables. But understand that it's the script convention.